### PR TITLE
Fttd 5

### PR DESCRIPTION
**Pull request by snaanortey**
Hi Ehimah,

Please see for review. Below is the summary of additions to the project:

1. Created a function `isActivityValid` to test the validity of `suggestedActivity` from a user; and
2. Saved `suggestedActivity` to a JSON file.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,6 +34,37 @@ boredAppServer.get("/random", function (request, response) {
...
+ 
+  // Add new suggested activity to an array
+  arrayOfSuggestedIdeas.push(suggestedActivity);
+
+  console.log(arrayOfSuggestedIdeas);
+
+  //Save the array of suggested activities to the suggestions file (suggestions.json)
+  const data = JSON.stringify(arrayOfSuggestedIdeas, null, 2);
+  const promise = writeFile("suggestions.json", data);
+  await promise;
```
Comment (ehimah) [inline index.js]: you can write this in one line like this
```suggestion
  await writeFile("suggestions.json", data);
```

File: index.js [hunk 2]
```diff
@@ -15,6 +34,37 @@ boredAppServer.get("/random", function (request, response) {
...
+
+  
+  response.send("Suggestion received, let me go and sleep sucker!");
```
Comment (ehimah) [inline index.js]: Mind your language!
```suggestion
  
  response.send("Suggestion received with many thanks 🤗!");
```

File: util.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Checks if the activity is valid
```
Comment (ehimah) [inline util.js]:
```suggestion
/**
 * Checks if the activity is valid and returns false if the object is not a valid activity
```

File: util.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  ) {
+    console.log("activity does not have a description property which is a string")
+    return false;
+  }
+  // Description property is not an empty string
+  if (activity.description.trim() === "") {
+    console.log("activity is an empty string")
+    return false;
+  }
+  return true;
```
Comment (ehimah) [inline util.js]: do you like the console logs as they are or do you want to remove them?

File: index.js [hunk 5]
```diff
@@ -15,6 +34,37 @@ boredAppServer.get("/random", function (request, response) {
...
+  const suggestedActivity = request.body;
+
+  // Calling the function to check that the object passed to the funtion is valid
+  const isValid = isActivityValid(suggestedActivity);
+
+  //Validate the input from the user and return bad request if the input is invalid
+  if (isValid === false) {
+    response.status(400).send("Your suggestion is invalid. Please try again");
+    return;
+  }
```
Comment (ehimah) [inline index.js]: neat 🔥

File: index.js [hunk 6]
```diff
@@ -1,11 +1,30 @@
 // Import express
 const express = require("express");
-//Import data
+
+// Import file system
+const { writeFile } = require("node:fs/promises");
+
+//Import data from JSON to be read in js
```
Comment (ehimah) [inline index.js]: you should maybe clean up imports -> Right click in `Visual Studio -> Organize imports`

**Review by ehimah — CHANGES_REQUESTED** [review]
This looks good. fix up these comments and we're good 👌

**Comment by snaanortey** [comment]
> 

Comments addressed. Please review.

**Review by ehimah — APPROVED** [review]
Good 2 GO! 🕺 

 ehimah-spin-colors

**Pull request merged**